### PR TITLE
Add notebook: Evaluating AI search engines with the judges library

### PR DESCRIPTION
**Pull request by jamesliounis**
---
### *Description*
This notebook showcases how to use *judges*—an open-source library for LLM-as-a-Judge evaluators—to assess and compare outputs from AI search engines like *Gemini*, *Perplexity*, and *EXA*.
---
### *What is judges?*
*judges* is an open-source library that provides researched-backed, ready-to-use LLM-based evaluators for assessing outputs across various dimensions such as correctness, quality, and harmfulness. It supports both:
- *Classifiers* (binary evaluations like True/False).
- *Graders* (scored evaluations on numerical scales).
The library also provides an integration with *litellm*, allowing access to most open- and closed-source models and providers.
---
### *What This Notebook Does*
1. Demonstrates how to use *judges* with *litellm* to evaluate AI search engine responses.
2. Uses *LLaMA 3* (together_ai/meta-llama/Llama-3.3-70B-Instruct-Turbo) as the LLM evaluator to assess:
   - *Correctness* (factual accuracy).
   - *Quality* (clarity, helpfulness).
3. Provides a step-by-step workflow to evaluate outputs generated by search engines.
---
### *Open-Source Tools & Resources*
- *Library*: judges
- *Model*: [LLaMA 3.3 70B Instruct-Turbo](https://huggingface.co/together_ai/meta-llama/Llama-3.3-70B-Instruct-Turbo) via litellm
- *Dataset*: [Natural Questions Subset](https://huggingface.co/datasets/quotientai/natural-qa-random-100-with-AI-search-answers)
---
### *Why This Notebook?*
This notebook provides a *practical example* of using *judges* with an open-source model (LLaMA 3) to evaluate real-world AI outputs. It highlights the library's flexibility, ease of integration with *litellm*, and usefulness for benchmarking AI systems in a transparent, reproducible manner.

### CODE REVIEW

**Comment by jamesliounis** [comment]
@merveenoyan @stevhliu

File: notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb [hunk 1]
```diff
@@ -0,0 +1,1687 @@
+{
```
Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>"<a href="https://github.com/quotient-ai/judges" target="_blank">judges</a> is an open-source library to use and create LLM-as-a-Judge evaluators. It provides a set of curated, research-backed..."</p><p> </p><p>"...a collection of real-world Google queries...as our benchmark for comparing..."</p><p> </p><p>"...which only includes human evaluated answers and their corresponding queries for correctness, clarity, and completeness."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>I'd move the table of contents to the very top</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>"...or from Google Colab secrets, in which case, uncomment the relevant code examples below."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Maybe add this to the Perplexity section?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Remove the last sentence</p><p> </p><p>"...reading in our <a href="https://huggingface.co/datasets/quotientai/natural-qa-random-67-with-AI-search-answers/tree/main/data" target="_blank">data</a> that now contains the search results."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (stevhliu) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Maybe clarify <code>MTBenchChatBotResponseQuality</code> is also a "grader" type of judge (not really clear right now). It can say something like "Response Quality Evaluation Grader" </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

**Review by stevhliu — COMMENTED** [review]
Very cool library! 👏 

Remember to add your notebook to the `toctree` and modify `index.md` to also include your notebook (remove one of the older notebooks from it and replace it with yours) to the latest notebooks [section](https://huggingface.co/learn/cookbook/index#latest-notebooks).

File: notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb (hunk 1 above)
Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Done!</p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Done!</p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Done!</p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Done!</p>

**Comment by jamesliounis** [comment]
Hey Stephen! Thanks for the prompt feedback. I have incorporated your comments, added nb to `toctree` and to `index.md`.

File: notebooks/en/index.md [hunk 2]
```diff
@@ -7,12 +7,12 @@ applications and solving various machine learning tasks using open-source tools
 
 Check out the recently added notebooks:
 
+- [Evaluating AI Search Engines with `judges` - the open-source library for LLM-as-a-judge evaluators](llm_judge_evaluating_ai_search_engines_with_judges_library)
 - [Multimodal Retrieval-Augmented Generation (RAG) with Document Retrieval (ColPali) and Vision Language Models (VLMs)](multimodal_rag_using_document_retrieval_and_vlms)
 - [Fine-Tuning a Vision Language Model (Qwen2-VL-7B) with the Hugging Face Ecosystem (TRL)](fine_tuning_vlm_trl)
 - [Multi-agent RAG System 🤖🤝🤖](multiagent_rag_system)
 - [Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs](multimodal_rag_using_document_retrieval_and_reranker_and_vlms)
 - [Fine-tuning SmolVLM with TRL on a consumer GPU](fine_tuning_smol_vlm_sft_trl)
-- [Smol Multimodal RAG: Building with ColSmolVLM and SmolVLM on Colab's Free-Tier GPU](multimodal_rag_using_document_retrieval_and_smol_vlm)
```
Comment (stevhliu) [inline notebooks/en/index.md:15]: Sorry I wasn't clear, we should keep the most recent ones (towards the bottom) and remove the one on top (`Multimodal Retrieval-Augmented Generation (RAG) with Document Retrieval (ColPali) and Vision Language Models (VLMs)`)

**Review by stevhliu — APPROVED** [review]
Cool, thanks! Once @merveenoyan has had a chance to review, we can merge :)

File: notebooks/en/index.md (hunk 2 above)
Comment (jamesliounis) [reply notebooks/en/index.md:15]: Modified it. Thanks for pointing that out.

**Comment by jamesliounis** [comment]
@merveenoyan Happy New Year! Have you had the chance to take a look?

File: notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb [hunk 3]
```diff
@@ -0,0 +1,1680 @@
...
```
Comment (merveenoyan) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>We use the&nbsp;<a href="https://paperswithcode.com/dataset/natural-questions" target="_blank">Natural Questions dataset</a>&nbsp;-- a collection of real-world Google queries and corresponding Wikipedia articles -- as our benchmark for comparing the quality of different AI search engines, as follows:</p><p> </p><p>this sentence is a bit too long and hard to follow, can we simplify it?</p><p> </p><p>nit: open-source* (there's an s at the end)</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (merveenoyan) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>it seems we already do this below, so no need to mention</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (merveenoyan) [inline notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>can you explain why you picked this instead of local serving? we often do local serving with open-source models in open-source cookbook </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/257/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

**Review by merveenoyan — COMMENTED** [review]
Left very minor nits, we can merge afterwards!
Sorry for the delay, I was off!

**Comment by freddiev4** [comment]
> Left very minor nits, we can merge afterwards! Sorry for the delay, I was off!

@merveenoyan 👋🏼 thanks for reviewing! I'm going shepherd this PR the rest of the way from our team. Will respond to your comments above and make updates + open a new PR if that's ok.

File: notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb (hunk 3 above)
Comment (freddiev4) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>We picked this instead of local serving because it's a bit more lightweight and users don't need to have a machine available with local serving set up to get started. We'd love to add support for that in <code>judges</code> though in the future!</p>

Comment (freddiev4) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>How about this?</p><p> </p><blockquote>We use the&nbsp;<a href="https://paperswithcode.com/dataset/natural-questions" target="_blank">Natural Questions dataset</a>&nbsp;as our benchmark for comparing the quality of different AI search engines. Natural Questions is a collection of real-world Google queries and corresponding Wikipedia articles. We'll walk through the following process:</blockquote>

Comment (freddiev4) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Noted :+1: !</p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Done!</p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Removed it.</p><p> </p>

Comment (jamesliounis) [reply notebooks/en/llm_judge_evaluating_ai_search_engines_with_judges_library.ipynb:1]: <p>Added a sentence to explain. </p>

**Comment by freddiev4** [comment]
@stevhliu @merveenoyan https://github.com/huggingface/cookbook/pull/270!
